runtime/src/accounts.rs

```diff
@@ -572,13 +568,26 @@ impl Accounts {
         txs: &[Transaction],
         txs_iteration_order: Option<&[usize]>,
     ) -> Vec<Result<()>> {
-        let keys: Vec<_> = OrderedIterator::new(txs, txs_iteration_order)
-            .map(|tx| tx.message().get_account_keys_by_lock_type())
+        use solana_sdk::sanitize::Sanitize;
+        let keys: Vec<Result<_>> = OrderedIterator::new(txs, txs_iteration_order)
+            .map(|tx| {
+                tx.sanitize()
+                    .map_err(|_| TransactionError::SanitizeFailure)?;
+
+                if Self::has_duplicates(&tx.message.account_keys) {
```

Comment:
why isn't this check part of sanitize?

Reply:
Yea, it could be. It's a bit more plumbing to translate to the AccountLoadedTwice error, but yea seems like it can go there. This was just the easiest place to move to.

Reply:
Up to you.  Not a blocker.  But it would be good to move this kind of logic out of the bank and have separate unit tests for this stuff in the sanitize implementation

Reply:
Yea, I'll do a follow-up change for that.